src/type-collector-snippet.ts

```diff
@@ -74,12 +74,32 @@ export function getTypeName(value: any, nest = 0): string | null {
     }
     if (value.constructor && value.constructor.name) {
         const { name } = value.constructor;
-        return name === 'Object' ? 'object' : name;
+        if (name === 'Object') {
+            return getObjectTypes(value, nest);
+        } else {
+            return name;
+        }
     }
 
     return typeof value;
 }
 
+function getObjectTypes(obj: any, nest: number): string {
+    const keys = Object.keys(obj);
```

Comment:
Let's sort the keys so we always get the same result for objects of the same shape.

e.g. given either of the following inputs:

```typescript
Object.keys({ a: 1, b : 2 }) //  ["a", "b"]
Object.keys({ b: 1, a : 2 }) // ["b", "a"]
```

we'd like to get the same output (let's also add a test case covering this)